File: data/src/main/java/com/somename/data/repository/datasource/DataSource.java
package com.somename.data.repository.datasource;

import com.somename.data.сontent.RootEntity;
import com.somename.data.сontent.RootProductEntity;
import com.somename.data.сontent.RootTopicsEntity;

import io.reactivex.Observable;

public interface DataSource {

    Observable<RootEntity> getTechCategoryEntity();

    Observable<RootTopicsEntity> getTrendingTopicsEntity();

    Observable<RootEntity> getChoosenTopicEntity(int topicId);

    Observable<RootProductEntity> getProductEntity(String id);

}

File: data/src/main/java/com/somename/data/repository/datasource/DataSource.kt
package com.somename.data.repository.datasource

import com.somename.data.сontent.RootEntity
import com.somename.data.сontent.RootProductEntity
import com.somename.data.сontent.RootTopicsEntity

import io.reactivex.Observable

interface DataSource {

    val techCategoryEntity: Observable<RootEntity>

    val trendingTopicsEntity: Observable<RootTopicsEntity>

    fun getChosenTopicEntity(topicId: Int): Observable<RootEntity>

    fun getProductEntity(id: String): Observable<RootProductEntity>

}
